source/exec/docker.d

```diff
@@ -95,9 +99,8 @@ class Docker: IExecProvider
 			return typeof(return)("Maximum number of parallel compiles has been exceeded. Try again later.", false);
 		}
 
-		++queueSize_;
-		scope(exit)
-			--queueSize_;
+		synchronized { ++queueSize_; }
+		scope(exit) synchronized { --queueSize_; }
```

Comment:
> The biggest problem however is that the synchronization is thread-wide meaning that it will stop the whole eventloop, essentially serving one user at a time. 

Yes, but that's a lot better than having a negative queueSize.

> I suggest you move the synchronization in the parallel section of the constructor.

It doesn't matter for the constructor - I could simply reset the `queueSize` afterwards

I will use CAS here instead then.